File: ignite-nest/src/post/entity/post.entity.ts
import { Column, Entity } from 'typeorm';
import { Base } from '../../common/entity/base';

@Entity('post')
export class Post extends Base {
  @Column({ type: 'varchar', length: 50, name: 'title' })
  title: string;

  @Column({ type: 'text', name: 'content' })
  content: string;

  @Column({ type: 'bigint', name: 'comment_count', default: 0 })
  commentCount: number;

  update(title?: string, content?: string) {
    this.title = title ?? this.title;
    this.content = content ?? this.content;
  }

  /**
   * 댓글 개수 차감시 value -1
   * @param value
   */
  updateCommentCount(value: number) {
    this.commentCount += value;
  }
}

File: ignite-spring/src/main/kotlin/org/rookedsysc/ignitespring/post/entity/PostEntity.kt
package org.rookedsysc.ignitespring.post.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import org.rookedsysc.ignitespring.common.entity.BaseEntity

@Entity
class PostEntity(title: String, content: String) : BaseEntity() {
    @Column(columnDefinition = "varchar(50)", length = 50, nullable = false)
    var title: String = ""
    @Column(columnDefinition = "text", nullable = false)
    var content: String = ""
    @Column(name = "comment_count", nullable = false)
    var commentCount: Int = 0
}
